lib/module.js

```diff
@@ -205,6 +205,9 @@ Module._findPath = function(request, paths, isMain) {
   var exts;
   var trailingSlash = request.length > 0 &&
                       request.charCodeAt(request.length - 1) === 47/*/*/;
+  if (!trailingSlash) {
+    trailingSlash = /(?:^|\/)\.?\.$/.test(request);
+  }
```

Comment:
Can you explain the bit of replacing the falsey `trailingSlash` result with another?
The regexp doesn't looks like it's testing for a trailing slash so the name is confusing.

Reply:
It would probably be clearer with another variable name

Reply:
this manually ensures using `/.` `/..` acts as a virtual trailing slash since it is doing a directory lookup see https://github.com/nodejs/node/issues/14990#issuecomment-324695609 .

Reply:
Should this account for other path separators than `/`?

Reply:
The specifier should have already been normalized by this point (hence only checking `/` above it, but I can't find where that is happening on a quick glance.